provider/pipeline_bake_stage.go

```diff
@@ -160,5 +169,16 @@ func (s *bakeStage) SetResourceData(d *schema.ResourceData) error {
 	if err != nil {
 		return err
 	}
-	return d.Set("vm_type", s.VMType)
+	err = d.Set("vm_type", s.VMType)
+	if err != nil {
+		return err
+	}
+
+	// Coveo specific
+	err = d.Set("pipeline_metadata", s.PipelineMetadata)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
(Nitpick) If you don't want yet another `if`, you can use the pattern that was previously there:
```suggestion
	return d.Set("pipeline_metadata", s.PipelineMetadata)
```
This will take care of returning `nil` or `not nil`